darts/models/regression_model.py

```diff
@@ -13,300 +13,289 @@
+                **kwargs
+            )
+        if self.max_lag == 0:
+            self.prediction_data = pd.DataFrame(columns=series.stack(other=exog).columns())
+        elif exog is not None:
+            self.prediction_data = series.stack(other=exog)[-self.max_lag:]
+        else:
+            self.prediction_data = series[-self.max_lag:]
 
-        raise_if_not(len(train_features) > 0, 'Need at least one feature series', logger)
-        raise_if_not(all([s.has_same_time_as(train_target) for s in train_features]),
-                     'All provided time series must have the same time index', logger)
-        self.train_features = train_features
-        self.train_target = train_target
         self._fit_called = True
 
-    @abstractmethod
-    def predict(self, features: List[TimeSeries]) -> TimeSeries:
-        """ Predicts values of the target time series, given a list of features time series
+    def _create_training_data(self, series: TimeSeries, exog: TimeSeries = None):
+        """ Create dataframe of exogenous and endogenous variables containing lagged values.
 
         Parameters
         ----------
-        features
-            The list of features time series, of the same length
+        series : TimeSeries
+            Target series.
+        exog : TimeSeries, optional
+            Exogenous variables.
 
         Returns
         -------
         TimeSeries
-            A series containing the predicted targets, of the same length as the features series
+            Data frame with lagged values.
         """
+        raise_if(series.width > 1,
+            "Series must not be multivariate. Pass exogenous variables to 'exog' parameter.",
+            logger
+        )
+        training_data_list = []
 
-        if (not self._fit_called):
-            raise_log(Exception('fit() must be called before predict()'), logger)
+        if self.lags is not None:
+            lagged_data = self._create_lagged_data(series=series, lags=self.lags)
+            training_data_list.append(lagged_data)
 
-        length_ok = len(features) == len(self.train_features)
-        dimensions_ok = all(features[i].width == self.train_features[i].width for i in range(len(features)))
-        raise_if_not(length_ok and dimensions_ok,
-                     'The number and dimensionalities of all given features must correspond to those used for'
-                     ' training.', logger)
+        if self.lags_exog is not None:
+            for i, col in enumerate(exog.columns()):
+                lagged_data = self._create_lagged_data(
+                    series=exog[col], lags=self.lags_exog
+                )
+                training_data_list.append(lagged_data)
 
-    def _historical_forecasts_sanity_checks(self, *args: Any, **kwargs: Any) -> None:
-        """Sanity checks for the historical_forecasts function
+        training_data = pd.concat(training_data_list, axis=1)
+        return TimeSeries(training_data.dropna(), freq=series.freq())
```

Comment:
Hm I could work around this by dropping the first `self.max_lags` values. Then probably most of the models down the road will complain. But this is probably better than silently removing `na`.

Reply:
I guess I don't even really need it at this place anymore because I'm calling the method `_create_lagged_data` from here. I'll handle it in there and just drop the first few rows.

Reply:
So I changed the above two lines in question to:
```python
training_data = pd.concat(training_data_list, axis=1)[self.max_lag:]
return TimeSeries(training_data, freq=series.freq())
```
Now the first few rows are dropped because there aren't enough lags available but I keep all other nans. I also introduced a new test for this.